digital_asset_types/src/dao/scopes/asset.rs

```diff
@@ -427,6 +429,83 @@ pub async fn get_by_id(
     })
 }
 
+pub async fn fetch_transactions(
+    conn: &impl ConnectionTrait,
+    tree: Vec<u8>,
+    leaf_idx: i64,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<(String, String)>, DbErr> {
+    let stmt = cl_audits_v2::Entity::find()
+        .filter(cl_audits_v2::Column::Tree.eq(tree))
+        .filter(cl_audits_v2::Column::LeafIdx.eq(leaf_idx))
+        .order_by(cl_audits_v2::Column::Seq, sea_orm::Order::Asc);
+
+    let stmt = paginate(
+        pagination,
+        limit,
+        stmt,
+        sea_orm::Order::Asc,
+        asset::Column::Id,
+    );
+    let transactions = stmt.all(conn).await?;
+    let transaction_list = transactions
+        .into_iter()
+        .map(|transaction| {
+            let tx = bs58::encode(transaction.tx).into_string();
+            let ix = Instruction::to_str(&transaction.instruction).to_string();
+            (tx, ix)
+        })
+        .collect();
+
+    Ok(transaction_list)
+}
+
+pub async fn get_asset_signatures(
+    conn: &impl ConnectionTrait,
+    asset_id: Option<Vec<u8>>,
+    tree_id: Option<Vec<u8>>,
+    leaf_idx: Option<i64>,
+    pagination: &Pagination,
+    limit: u64,
+) -> Result<Vec<(String, String)>, DbErr> {
+    // if tree_id and leaf_idx are provided, use them directly to fetch transactions
+    if let (Some(tree_id), Some(leaf_idx)) = (tree_id, leaf_idx) {
+        let transactions = fetch_transactions(conn, tree_id, leaf_idx, pagination, limit).await?;
+        return Ok(transactions);
+    }
+
+    if asset_id.is_none() {
+        return Err(DbErr::Custom(
+            "Either 'id' or both 'tree' and 'leafIndex' must be provided".to_string(),
+        ));
+    }
+
+    // if only asset_id is provided, fetch the latest tree and leaf_idx (asset.nonce) for the asset
+    // and use them to fetch transactions
+    let stmt = asset::Entity::find()
+        .distinct_on([(asset::Entity, asset::Column::Id)])
+        .filter(asset::Column::Id.eq(asset_id))
+        .order_by(asset::Column::Id, Order::Desc)
+        .limit(1);
```

Comment:
getSignaturesForAddress provided by vanilla solana PRC orders the transactions by desc slot. If its desired to mimic that call but for compression trees then should order by seq.

Reply:
index on seq should be added if order is switched.

Reply:
This is fetching the asset to lookup the tree and leaf info. You actually don't even need the `order_by` since asset_id is a unique field.

Reply:
You'll see that `fetch_transactions` orders by Seq ascending. So the opposite direction of `getSignauresForAddress`. 

In hindsight, doing DESC would have been better. Ideally I'd rather not change it now to avoid breaking our customers. If reviewers feel strongly about it we can discuss with our customers, too.

Reply:
If its a single or minimal amount of customers who are impacted by the ordering, might be worth asking them quick if they depend on the order and if it would break their use case.

I think ideally we would match `getSignaturesForAddress` if there's no or very minimal impact.  I agree if it breaks them or is some work, we don't want to change it.  But if you could just ask them and they say it doesn't matter then we should change it imo.

Reply:
To reiterate this comment is not a blocker just a request if its quick to do.